sdks/wasm/framework/src/WebAssembly.Net.WebSockets/WebSocketHelpers.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Globalization;
+using System.Threading;
+
+namespace WebAssembly.Net.WebSockets {
+	internal static class WebSocketHelpers {
+
+		private const string Separators = "()<>@,;:\\\"/[]?={} ";
+
+		private static readonly ArraySegment<byte> s_EmptyPayload = new ArraySegment<byte> (new byte [] { }, 0, 0);
+
+		internal static ArraySegment<byte> EmptyPayload {
+			get { return s_EmptyPayload; }
+		}
+
+
+
+		internal static void ValidateSubprotocol (string subProtocol)
+		{
+			if (string.IsNullOrWhiteSpace (subProtocol)) {
+				throw new ArgumentException ("Protocol string can not be null or empty", nameof (subProtocol));
+			}
+
+			char [] chars = subProtocol.ToCharArray ();
+			string invalidChar = null;
+			int i = 0;
+			while (i < chars.Length) {
+				char ch = chars [i];
+				if (ch < 0x21 || ch > 0x7e) {
+					invalidChar = string.Format (CultureInfo.InvariantCulture, "[{0}]", (int)ch);
+					break;
+				}
+
+				if (!char.IsLetterOrDigit (ch) &&
+				    Separators.IndexOf (ch) >= 0) {
+					invalidChar = ch.ToString ();
+					break;
+				}
+
+				i++;
+			}
+
+			if (invalidChar != null) {
+				throw new ArgumentException ($"Invalid char: {invalidChar} in Protocol string");
+			}
+		}
+
+		internal static void ValidateOptions (string subProtocol,
+		    int receiveBufferSize,
+		    int sendBufferSize,
+		    TimeSpan keepAliveInterval)
+		{
+			// We allow the subProtocol to be null. Validate if it is not null.
+			if (subProtocol != null) {
+				ValidateSubprotocol (subProtocol);
+			}
+
+			if (keepAliveInterval < Timeout.InfiniteTimeSpan) // -1
+			{
+				throw new ArgumentOutOfRangeException (nameof (keepAliveInterval), keepAliveInterval,
+				    $"Argument specified '{Timeout.InfiniteTimeSpan.ToString ()}' is too small.");
```

Comment:
Shouldn't the 'argument specified' bit be {keepAliveInterval} and not InfiniteTimeSpan?

Reply:
good catch.  Recent commits now take this validation out so it should be resolved.